Core-cardmodule fix

kds1660 opened the pull request:
1. Fixed in cardmodule error in setNewBalance method
2. Add in core setNewBalance after cash retrieving
3. Change - now working card date check and add in core

kds1660 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into core

kds1660 committed:
changes

1. Fixed in cardmodule error in setNewBalance method
2. Add in core setNewBalance after cash retrieving
3. Change - now working card date check and add in core

kds1660 committed:
cardModule test

i don't understend why isFalse was, change to isTrue if pin and date
correct

Varkh commented on js/controller/core.js:
```diff
@@ -200,6 +202,7 @@ function Core(cashModule, cardModule, navigation,cashOutModule) {
                         var cashOut = cashModule.getCash(+cash);
                         navigation.showMessage(STATE_TEXT.ALL_OK);
                         cashOutModule.showMoney(cashOut);
+                        cardModule.setNewBalance(+cash);
```
I would prefer `parseInt` function

Varkh commented on js/model/cardDataModel.js:
```diff
@@ -67,7 +67,7 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
      */
     this.setBalance = function (newBallance) {
-        this.ballance = newBallance;
+        this._ballance = newBallance;
```
I think it's better to have `balance` as private field(not in this)

kds1660 replied:
first we must implement traffic card in atm, for now  - we creating new card for every step, i guess - in main.js create 3 objects-cards and working with it - push in atm and core must back this objects ande change event card inserted - in details object, in core add event card-ejected and details - card object. something like this

Varkh replied:
Leave all card logic in card module. Only card module should handle CardModel

Varkh commented on js/model/cardDataModel.js:
```diff
@@ -87,13 +87,17 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
 
 CardDataModel.prototype.isNotExpired = function () {
     var now = new Date();
```
Create `expirationDate` with `new Date`
and just check
```
now.getTime() < expirationDate.getTime()
```

kds1660 committed:
changes

1) from displayViev move to css some code
2) add changePin to CardModule
3)add changePin to menu
4)change '+' to parseInt
5) change check date method
6) changer for to each in displayView

kds1660 committed:
implemet basic card move in atm

1.card1 in main, move into core and out of this
2. core fixed errors

Varkh commented:
Tests are failing

kds1660 committed:
fix tests

Varkh commented on css/main.css:
```diff
@@ -131,4 +131,20 @@ p {
\ No newline at end of file
+}
+
+table {
```
don't use styles for elements, use classes
what if one more table will be added

kds1660 commented:
var event = new CustomEvent не прошел, пока закомменил


2016-12-07 22:08 GMT+02:00 Denys <notifications@github.com>:

> Tests are failing
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Varkh/CashMachine-JS-Emulator/pull/65#issuecomment-265558849>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AV8voJFf-qKjJ8iL0o32Jrxcw11C_HASks5rFxIlgaJpZM4LGiav>
> .
>

Varkh commented on js/model/cardDataModel.js:
```diff
@@ -60,14 +62,14 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
 
     this.getBallance = function () {
-        return ballance;
+        return self._ballance;
```
not good.
Was better before

kds1660 replied:
i dont understend for now - this - div with value 500, not find where is this happens, for now change to self

Varkh commented on js/model/cardDataModel.js:
```diff
@@ -87,13 +89,10 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
-        && now.getMonth() < this._expirationDate[1]
-        && now.getDate() < this._expirationDate[0]) {
+    var expDate = new Date (this._expirationDate[2],this._expirationDate[1]-1,this._expirationDate[0])
+    if(now.getTime() < expDate.getTime()) {
```
Looks better now :)

Varkh reviewed (approved):
Approved but changes required

Varkh commented on js/model/cardDataModel.js:
```diff
@@ -8,6 +8,7 @@
  * @constructor
 **/
 function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
+    var self=this;
```
remove self

Varkh commented on js/model/cardDataModel.js:
```diff
@@ -60,14 +62,14 @@ function CardDataModel(holderName, cardNumber, expirationDate, ballance) {
     };
 
     this.getBallance = function () {
-        return ballance;
```
I want to see:

`var ballance;`
not
`this._ballance;`

kds1660 reviewed (commented):
i remember, for first I implement card move in atm, now try remove this._balanse and this remedy situation with self in cardmodel, i guess
